tests/db_engine_specs/superset_tests.py

```diff
@@ -0,0 +1,159 @@
+    session.commit()
+
+    yield database
+
+    session.delete(database)
+    session.commit()
+    os.unlink("database2.db")
+
+
+@pytest.fixture()
+def table2(database2):
+    engine = database2.get_sqla_engine()
+    conn = engine.connect()
+    conn.execute("CREATE TABLE table2 (a INTEGER NOT NULL PRIMARY KEY, b TEXT)")
+    conn.execute("INSERT INTO table2 (a, b) VALUES (1, 'ten'), (2, 'twenty')")
+    session.commit()
+
+    yield
+
+    conn.execute("DROP TABLE table2")
+    session.commit()
+
+
+@mock.patch("superset.security.manager.g")
+def test_superset(g, app_context, table1):
+    g.user = security_manager.find_user("admin")
+
+    engine = create_engine("superset://")
+    conn = engine.connect()
+    results = conn.execute('SELECT * FROM "superset.database1.table1"')
+    assert list(results) == [(1, 10), (2, 20)]
+
+
+@mock.patch("superset.security.manager.g")
+def test_superset_joins(g, app_context, table1, table2):
+    g.user = security_manager.find_user("admin")
+
+    engine = create_engine("superset://")
+    conn = engine.connect()
+    results = conn.execute(
+        """
+        SELECT t1.b, t2.b
+        FROM "superset.database1.table1" AS t1
+        JOIN "superset.database2.table2" AS t2
+        ON t1.a = t2.a
+        """
+    )
+    assert list(results) == [(10, "ten"), (20, "twenty")]
+
+
+@mock.patch("superset.security.manager.g")
+def test_dml(g, app_context, table1, table2):
+    g.user = security_manager.find_user("admin")
+
+    engine = create_engine("superset://")
+    conn = engine.connect()
+
+    conn.execute('INSERT INTO "superset.database1.table1" (a, b) VALUES (3, 30)')
+    results = conn.execute('SELECT * FROM "superset.database1.table1"')
+    assert list(results) == [(1, 10), (2, 20), (3, 30)]
+    conn.execute('UPDATE "superset.database1.table1" SET b=35 WHERE a=3')
+    results = conn.execute('SELECT * FROM "superset.database1.table1"')
+    assert list(results) == [(1, 10), (2, 20), (3, 35)]
+    conn.execute('DELETE FROM "superset.database1.table1" WHERE b>20')
+    results = conn.execute('SELECT * FROM "superset.database1.table1"')
+    assert list(results) == [(1, 10), (2, 20)]
+
+    with pytest.raises(ProgrammingError) as excinfo:
+        conn.execute(
+            """INSERT INTO "superset.database2.table2" (a, b) VALUES (3, 'thirty')"""
+        )
+    assert (
+        str(excinfo.value).strip()
+        == '(shillelagh.exceptions.ProgrammingError) DML not enabled in database "database2"\n[SQL: INSERT INTO "superset.database2.table2" (a, b) VALUES (3, \'thirty\')]\n(Background on this error at: http://sqlalche.me/e/13/f405)'
+    )
+
+
+@mock.patch("superset.security.manager.g")
+def test_security_manager(g, app_context, table1):
+    g.user = security_manager.find_user("gamma")
+
+    engine = create_engine("superset://")
+    conn = engine.connect()
+    with pytest.raises(SupersetSecurityException) as excinfo:
+        conn.execute('SELECT * FROM "superset.database1.table1"')
```

Comment:
We have database, schema and dataset level access. Can you add those test combinations also

Reply:
Will do!

Reply:
@dpgaspar do you mean adding roles with DB-level permissions and testing this? Same for schema and dataset?

Can you point me to examples? I took a look at our tests but couldn't find anything like this.

Reply:
Boy, has it almost been a year?

@dpgaspar the adapter is simply calling:

```
security_manager.raise_for_access(
    database=database,
    table=sql_parse.Table(self.table, self.schema, self.catalog),
    username=self.username,
)
```

So I think we can rely on existing tests for the security manager.